source/modulo_components/modulo_components/component_interface.py

```diff
@@ -390,11 +390,14 @@ def __subscription_callback(self, message: MsgT, attribute_name: str, reader: Ca
         except (AttributeError, MessageTranslationError) as e:
             self.get_logger().warn(f"Failed to read message for attribute {attribute_name}: {e}",
                                    throttle_duration_sec=1.0)
+        try:
+            user_callback()
+        except Exception as e:
+            self.get_logger().error(f"Failed to execute user callback in subscription for attribute"
+                                    f" '{attribute_name}': {e}", throttle_duration_sec=1.0)
```

Comment:
To be consistent with C++ behaviour, the user callback should not be invoked if the message reading fails. There is a test for this `test_component_interface.cpp`. Could you check that the same case is also tested in Python?

```suggestion
        except (AttributeError, MessageTranslationError) as e:
            self.get_logger().warn(f"Failed to read message for attribute {attribute_name}: {e}",
                                   throttle_duration_sec=1.0)
            return

        try:
            user_callback()
        except Exception as e:
            self.get_logger().error(f"Failed to execute user callback in subscription for attribute"
                                    f" '{attribute_name}': {e}", throttle_duration_sec=1.0)
```

Reply:
Indeed. While I was adding that test, I realized that the reader for an EncodedState message does not check for the message type, i.e. a component that is expecting a Cartesian state will not give an error in the subscription callback if it receives a joint state..so it will be the same for all other combinations..Do we want this to stay like that?